### PR TITLE
Leaner MacOS tests (selective installation of prerequisites).

### PR DESCRIPTION
**Pull request by jtattermusch**
Install only the prerequisites needed in prepare_build script.
(installing prerequisites for some languages generates lots of logs and it takes minutes or even more).

### CODE REVIEW

**Comment by jtattermusch** [comment]
@lidizheng  can you take an initial look?

**Comment by jtattermusch** [comment]
> I think this effort is great, save time and resources. Current solution definitely works. I want to discuss the implementation alternatives.
> 
> 1. If we are going to use environment variables to dictate whether to install certain dependencies, can we set those variables in `.cfg` files as `env_vars`? It is more cohesive and structure.

We could set them in the .cfg files as env_vars but one disadvantage is that for most jobs, there's two .cfg files - one for the master job (continuous runs) and one for PRs. With that setup, we'd need to duplicate the env_vars setting in both of the .cfg files and that seems very fragile (would most likely end up with failures on master tests pretty soon).

> 2. If we want to setup build environment in actual build scripts, is it possible to use command line argument than environment variables? So those variables will only be effective for the installation script, and not out-living that script.

For some reason I'm not a fan of the cmdline args (usually the places where we use extra args in our build scripts feel less readable), so I'd prefer to you the env vars. The names of the env vars are chosen to be fairly specific, so it's very unlikely they'd cause unexpected problems somewhere.

**Comment by jtattermusch** [comment]
@muxi  FYI

File: tools/internal_ci/helper_scripts/prepare_build_macos_rc [hunk 1]
```diff
@@ -27,62 +34,83 @@ if [ -n "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ]; then
...
-time git submodule update --init
+  # set xcode version for Obj-C tests
+  sudo xcode-select -switch /Applications/Xcode_9.2.app/Contents/Developer/
+
+  # Needed for ios-binary-size
+  time pip install --user pyyaml pyjwt cryptography requests
+
+  # Store intermediate build files of ObjC tests into /tmpfs
+  # TODO(jtattermusch): add explanation why this is needed
+  mkdir /tmpfs/Build-ios-binary-size
```
Comment (jtattermusch) [inline tools/internal_ci/helper_scripts/prepare_build_macos_rc]: @muxi   do you know why is this step needed for the ios-binary-size build?

**Comment by jtattermusch** [comment]
This is now ready for review.

File: tools/internal_ci/helper_scripts/prepare_build_macos_rc (hunk 1 above)
Comment (muxi) [reply tools/internal_ci/helper_scripts/prepare_build_macos_rc]: When we wrote the ios binary size test, Kokoro was running out of space during building. An instruction from Kokoro team was to move the build intermediates into `/tmpfs` that has more space. So this directory was made to store those intermediate files.

File: tools/internal_ci/helper_scripts/prepare_build_macos_rc [hunk 2]
```diff
@@ -27,62 +34,83 @@ if [ -n "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ]; then
...
 
-# Store intermediate build files of ObjC tests into /tmpfs
-mkdir /tmpfs/Build-ios-binary-size
-ln -s /tmpfs/Build-ios-binary-size src/objective-c/examples/Sample/Build
-mkdir -p /tmpfs/DerivedData
-rm -rf ~/Library/Developer/Xcode/DerivedData
-mkdir -p ~/Library/Developer/Xcode
-ln -s /tmpfs/DerivedData ~/Library/Developer/Xcode/DerivedData
+  # make sure md5sum is available
+  brew install coreutils
```
Comment (lidizheng) [inline tools/internal_ci/helper_scripts/prepare_build_macos_rc]: This line seems make the presubmit test unhappy. https://source.cloud.google.com/results/invocations/24a6a26e-8bb6-47e8-b138-d4af394b04a2

**Comment by lidizheng** [comment]
> We could set them in the .cfg files as env_vars but one disadvantage is that for most jobs, there's two .cfg files - one for the master job (continuous runs) and one for PRs. With that setup, we'd need to duplicate the env_vars setting in both of the .cfg files and that seems very fragile (would most likely end up with failures on master tests pretty soon).

Good to know that the cfg is usually duplicated for master and PRs.

> For some reason I'm not a fan of the cmdline args (usually the places where we use extra args in our build scripts feel less readable), so I'd prefer to you the env vars. The names of the env vars are chosen to be fairly specific, so it's very unlikely they'd cause unexpected problems somewhere.

Command line arguments indicate direct relationship between the parent script and child script, also help reader trace the variable preparing logic. For this use case, env var works, and if we keep the naming special, we shouldn't have any trouble.

File: tools/internal_ci/helper_scripts/prepare_build_macos_rc (hunk 2 above)
Comment (jtattermusch) [reply tools/internal_ci/helper_scripts/prepare_build_macos_rc]: I'll try to fix it.

**Pull request merged**